packages/material-ui-styles/src/withStyles/withStyles.d.ts

```diff
@@ -75,8 +76,9 @@ export interface StyledComponentProps<ClassKey extends string = string> {
 }
 
 export default function withStyles<
-  S extends Styles<any, any>,
-  Options extends WithStylesOptions = {}
+  S extends Styles<Theme, any>,
+  Options extends WithStylesOptions<Theme> = {},
+  Theme = any
```

Comment:
@vitkon Needs the same treatment as `makeStyles`.

Reply:
@eps1lon , all done. Have a look hen you have a min.

Reply:
The generic `Theme` argument should be removed and usage replaced with `ThemeOfStyles`

Reply:
erm... probably push didn't go through

Reply:
done 😥